### PR TITLE
[Enhanced Http] Cleanup

### PR DESCRIPTION
**Pull request by satvu**
### Issue describing the changes in this PR

Cleanup of the feature branch - removed changes in core that are not being used (left over from different prototyping stages).

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: extensions/Worker.Extensions.Http.AspNetCore/src/HostBuilderExtensions.cs [hunk 1]
```diff
@@ -22,7 +24,7 @@ public static class HostBuilderExtensions
         public static IHostBuilder ConfigureAspNetCoreIntegration(this IHostBuilder builder)
         {
             // TODO: Update this logic. Port should always come through configuration.
-            var port = Environment.GetEnvironmentVariable("Azure_Functions_HttpProxyingPort") ?? "5555";
+            var port = Environment.GetEnvironmentVariable("AZURE_FUNCTIONS_HTTP_PROXY_PORT") ?? "5555";
```
Comment (satvu) [inline extensions/Worker.Extensions.Http.AspNetCore/src/HostBuilderExtensions.cs:27]: this logic will be updated later (port should always come through config) but @brettsam is this name better?

Comment (satvu) [reply extensions/Worker.Extensions.Http.AspNetCore/src/HostBuilderExtensions.cs:27]: Related change in host feature branch in [this PR](https://github.com/Azure/azure-functions-host/pull/9149/files).

Comment (brettsam) [reply extensions/Worker.Extensions.Http.AspNetCore/src/HostBuilderExtensions.cs:27]: We should start with `FUNCTIONS_` -- I didn't realize that we have a bunch of env vars that use that prefix already, so let's stick with it.

**Review by brettsam — APPROVED** [review]

File: samples/FunctionApp/FunctionApp.csproj [hunk 2]
```diff
@@ -13,8 +13,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Azure.Functions.Worker" Version="1.13.0-preview5-20230303.1" />
```
Comment (liliankasem) [inline samples/FunctionApp/FunctionApp.csproj:16]: I think you still need a worker package reference here (latest is 1.10.1) or are you going to use a project reference?

**Review by liliankasem — COMMENTED** [review]
+1 to what Brett said :)

**Review by liliankasem — APPROVED** [review]

File: samples/FunctionApp/FunctionApp.csproj (hunk 2 above)
Comment (satvu) [reply samples/FunctionApp/FunctionApp.csproj:16]: the project reference is below in the file - we don't have a preview package out for this feature yet so not sure what this one is referencing tbh.

**Pull request merged**